packages/salesforcedx-vscode-core/src/orgBrowser/orgMetadata.ts

```diff
@@ -0,0 +1,157 @@
+    super();
+    this.outputPath = outputPath;
+  }
+
+  public build(data: {}): Command {
+    return new SfdxCommandBuilder()
+      .withArg('force:mdapi:describemetadata')
+      .withJson()
+      .withFlag('-f', this.outputPath)
+      .withLogName('force_describe_metadata')
+      .build();
+  }
+
+  public execute(response: ContinueResponse<string>): void {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async data => {
+      this.logMetric(execution.command.logName, startTime);
+      buildTypesList(this.outputPath);
+    });
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+    channelService.streamCommandOutput(execution);
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+const workspaceChecker = new SfdxWorkspaceChecker();
+const parameterGatherer = new EmptyParametersGatherer();
+
+export async function forceDescribeMetadata(outputPath?: string) {
+  if (isNullOrUndefined(outputPath)) {
+    outputPath = await getMetadataTypesPath();
+  }
+  const describeExecutor = new ForceDescribeMetadataExecutor(outputPath!);
+  const commandlet = new SfdxCommandlet(
+    workspaceChecker,
+    parameterGatherer,
+    describeExecutor
+  );
+  await commandlet.run();
+}
+
+export async function getMetadataTypesPath(): Promise<string | undefined> {
+  if (hasRootWorkspace()) {
+    const workspaceRootPath = getRootWorkspacePath();
+    const defaultUsernameOrAlias = await OrgAuthInfo.getDefaultUsernameOrAlias();
+    const defaultUsernameIsSet = typeof defaultUsernameOrAlias !== 'undefined';
+
+    if (defaultUsernameIsSet) {
+      const username = await OrgAuthInfo.getUsername(defaultUsernameOrAlias!);
+      const metadataTypesPath = path.join(
+        workspaceRootPath,
+        '.sfdx',
+        'orgs',
+        username,
+        'metadata',
+        'metadataTypes.json'
+      );
+      return metadataTypesPath;
+    } else {
+      const err = nls.localize('error_no_default_username');
+      telemetryService.sendError(err);
+      throw new Error(err);
+    }
+  } else {
+    const err = nls.localize('cannot_determine_workspace');
+    telemetryService.sendError(err);
+    throw new Error(err);
+  }
+}
+
+export type MetadataObject = {
+  directoryName: string;
+  inFolder: boolean;
+  metaFile: boolean;
+  suffix: string;
+  xmlName: string;
+};
+
+export function buildTypesList(metadataTypesPath: string): string[] {
+  if (fs.existsSync(metadataTypesPath)) {
```

Comment:
Since we're parsing a json, I think we should wrap the method on a try catch that surfaces that throws the error and have the code calling this handle surfacing the message.

Reply:
Okay, that works